sentry-android-core/src/main/java/io/sentry/android/core/util/ConnectivityChecker.java

```diff
@@ -35,20 +37,81 @@ private ConnectivityChecker() {}
+      return null;
+    }
+    return activeNetwork.isConnected();
+  }
+
+  @SuppressWarnings({"deprecation", "MissingPermission"})
+  private static @Nullable android.net.NetworkInfo getActiveNetworkInfo(
+      @NotNull Context context,
+      @NotNull ConnectivityManager connectivityManager,
+      @NotNull ILogger logger) {
     if (!Permissions.hasPermission(context, Manifest.permission.ACCESS_NETWORK_STATE)) {
       logger.log(SentryLevel.INFO, "No permission (ACCESS_NETWORK_STATE) to check network status.");
       return null;
     }
 
-    // do not import class or deprecation lint will throw
-    android.net.NetworkInfo activeNetwork = connectivityManager.getActiveNetworkInfo();
+    return connectivityManager.getActiveNetworkInfo();
+  }
 
-    if (activeNetwork != null) {
-      return activeNetwork.isConnected();
+  /**
+   * Check the connection type of the active network
+   *
+   * @param context the App. context
+   * @param logger the logger from options
+   * @return the connection type wifi, ethernet, cellular or null
+   */
+  @SuppressLint({"ObsoleteSdkInt", "MissingPermission"})
+  public static String getConnectionType(@NotNull Context context, @NotNull ILogger logger) {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+      ConnectivityManager connectivityManager = getConnectivityManager(context, logger);
+      if (connectivityManager == null) {
+        return null;
+      }
+      if (!Permissions.hasPermission(context, Manifest.permission.ACCESS_NETWORK_STATE)) {
+        logger.log(
+            SentryLevel.INFO, "No permission (ACCESS_NETWORK_STATE) to check network status.");
+        return null;
+      }
+      Network activeNetwork = connectivityManager.getActiveNetwork();
+      if (activeNetwork == null) {
+        logger.log(SentryLevel.INFO, "Network is null and cannot check network status");
+        return null;
+      }
+      NetworkCapabilities networkCapabilities =
+          connectivityManager.getNetworkCapabilities(activeNetwork);
+      if (networkCapabilities == null) {
+        logger.log(SentryLevel.INFO, "NetworkCapabilities is null and cannot check network type");
+        return null;
+      }
+      if (networkCapabilities.hasTransport(NetworkCapabilities.TRANSPORT_WIFI)) {
+        return "wifi";
+      }
+      if (networkCapabilities.hasTransport(NetworkCapabilities.TRANSPORT_ETHERNET)) {
+        return "ethernet";
+      }
+      if (networkCapabilities.hasTransport(NetworkCapabilities.TRANSPORT_ETHERNET)) {
+        return "cellular";
+      }
     }
-    logger.log(SentryLevel.INFO, "NetworkInfo is null and cannot check network status");
     return null;
   }
+
+  private static @Nullable ConnectivityManager getConnectivityManager(
+      @NotNull Context context, @NotNull ILogger logger) {
+    ConnectivityManager connectivityManager =
+        (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
```

Comment:
Can this cast ever throw?

Reply:
no, because of `Context.CONNECTIVITY_SERVICE`